### DISCUSSION TITLE
[✨] ErrorBoundary

### DISCUSSION THREAD

**Discussion by mhevery** [QwikDev/qwik-evolution#161] [Proposals For Qwik]
### Is your feature request related to a problem?

I would like to be able to intercept errors and render custom output.
- If an error happens in a child component, the parent component should be able to intercept it to render a custom error for that component branch only. (Think about how this plays with SSR)

### Describe the solution you'd like

```tsx
import { component$, useErrorBoundary } from '@builder.io/qwik';
import { ErrorBoundary } from '@builder.io/qwik-city';

export default component$(() => {
  const error = useErrorBoundary();
  return (
    <ErrorBoundary onError={(e) => <>ERROR: {e}</> }>
      <ProblemComponent />
    </ErrorBoundary>
  );
});

export const ProblemComponent = component$(() => {
  throw new Error('Component which throws error');
});
```

### Describe alternatives you've considered

Qwik does not currently provide anything like this.

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
